### PR TITLE
feat: resolve associated effects in EffUnification

### PR DESCRIPTION
**Pull request by mlutze**
extracted from #6499

### CODE REVIEW

**Comment by magnus-madsen** [comment]
!!!jar

File: main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification.scala [hunk 1]
```diff
@@ -127,26 +129,100 @@ object EffUnification {
...
+          lookupOrSolve(t1, t2, renv)
+        }
+
+        substRes.map {
+          case subst0 =>
+            // replace the temporary variables with the assoc types they represent
+            val subst1 = Substitution(backSubst) @@ subst0
+            // remove the temporary variables from the substitution
+            val subst = backSubst.keys.foldLeft(subst1)(_.unbind(_))
+            (subst, Nil)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification.scala:178]: Would we want to remove some bindings from subst? (e.g. the bindings for variables we no longer care about).

**Review by magnus-madsen — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification.scala (hunk 1 above)
Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification.scala:178]: Isn't that what line 177 does?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification.scala:178]: Aaah. Something about forests and trees.

**Pull request merged**